### PR TITLE
change rounding mode in theano backend to match tensorflow backend

### PR DESCRIPTION
**Pull request by yhenon**
Theano uses round away from zero, while tensorflow uses round to nearest even number.
Example to demonstrate the issue:

	class Rounder(Layer):

		def __init__(self, **kwargs):
			super(Rounder, self).__init__(**kwargs)

		def get_output_shape_for(self, input_shape):
			return (input_shape[0], input_shape[1])

		def call(self, x, mask=None):
			x1 = K.round(x)
			return x1

	model = Sequential()
	model.add(Rounder(input_shape=(4,)))
	model.compile(loss='mse', optimizer='sgd')
	A = model.predict(np.array([-0.5,0.5,1.5,2.5]).reshape(1,-1))
	print(A)

Output with tensorflow:

     [[ 0.  0.  2.  2.]]

Output with theano:

    [[-1.  1.  2.  3.]]

Output with theano after fix:

    [[-0.  0.  2.  2.]]
This night seem minor, but it introduces issues when trying to match behaviour from botch backends when rounding numbers that end in .5 (note that even when working with 32bit floats, many numbers of this form can be represented with full precision).

This pull request changes the rounding mode in Theano to match the rounding mode in tensorflow, and updates the documentation to clarify the rounding mode used.

I previously created issue #5076 that this request fixes.

### CODE REVIEW

**Comment by fchollet** [comment]
LGTM, thanks.

**Comment by nouiz** [comment]
Thanks for catching this. Theano should have the same default as NumPy. But for unknow reason, we don't in this case.

I made a PR to Theano to fix this: https://github.com/Theano/Theano/pull/5437

This PR to keras will continue to be good and will remove Theano warning about this. Which is probably what you want to do.
